src/bidiMapper/domains/context/browsingContextImpl.ts

```diff
@@ -407,10 +404,12 @@ export class BrowsingContextImpl {
               },
               this.id
             );
+            this.#deferreds.Page.lifecycleEvent.DOMContentLoaded.resolve(
```

Comment:
We should first send the event then we should resolve the differed. That way we insure that we have the following behavior:
Navigation request `completed` -> DomContentLoaded -> Loaded -> Navigation finished